packages/next/src/client/components/router-reducer/create-optimistic-tree.ts

```diff
@@ -49,5 +51,7 @@ export function createOptimisticTree(
     result[3] = 'refetch'
   }
 
+  result[4] = isRootLayout
```

Comment:
diff this obj (set `prefetch` to `true` and `false`): https://github.com/vercel/next.js/blob/fe8c43213df099602724bf2e2499564fcfbb1332/packages/next/src/client/components/router-reducer/reducers/navigate-reducer.ts#L32-L66

and compare the return type:
https://github.com/vercel/next.js/blob/fe8c43213df099602724bf2e2499564fcfbb1332/packages/next/src/server/app-render.tsx#L640-L649

then we will find `isRootLayout` is missing.